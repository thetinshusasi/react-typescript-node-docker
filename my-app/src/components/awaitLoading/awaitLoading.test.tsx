import React from "react";
import { shallow } from "enzyme";
import AwaitLoading from "./index";
import { Spinner } from "react-bootstrap";
// import { Button, Form } from "react-bootstrap";

describe("<AwaitLoading />", () => {
  it("renders AwaitLoading", () => {
    const content = <div>Test</div>;
    const component = shallow(<AwaitLoading>{content}</AwaitLoading>);
    expect(component).toMatchSnapshot();
  });
  it("renders AwaitLoading loading", () => {
    const content = <div>Test</div>;
    const component = shallow(
      <AwaitLoading loading={true}>{content}</AwaitLoading>
    );
    expect(component.find(Spinner).length).toEqual(1);
  });
  it("renders AwaitLoading error", () => {
    const content = <div>Test</div>;
    const component = shallow(
      <AwaitLoading error="error">{content}</AwaitLoading>
    );
    expect(component.find(".spinnerContainer").text()).toEqual("error");
  });
});
