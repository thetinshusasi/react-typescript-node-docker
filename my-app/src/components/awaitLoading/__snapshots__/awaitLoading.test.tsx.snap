// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTodo /> renders AddTodo 1`] = `
<Fragment>
  <Row
    noGutters={false}
  >
    <Col
      className="content"
      xs={12}
    >
      <div>
        Test
      </div>
    </Col>
  </Row>
</Fragment>
`;

exports[`<AddTodo /> renders AddTodo loading 1`] = `
<Fragment>
  <Row
    noGutters={false}
  >
    <Col
      className="spinnerContainer"
      xs={12}
    >
      <Spinner
        animation="border"
        variant="primary"
      />
    </Col>
  </Row>
</Fragment>
`;

exports[`<AwaitLoading /> renders AwaitLoading 1`] = `
<Fragment>
  <Row
    noGutters={false}
  >
    <Col
      className="content"
      xs={12}
    >
      <div>
        Test
      </div>
    </Col>
  </Row>
</Fragment>
`;
